src/parser/shaman/elemental/modules/azerite/LavaShock.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'parser/core/Analyzer';
+import { formatNumber } from 'common/format';
+import TraitStatisticBox, { STATISTIC_ORDER } from 'interface/others/TraitStatisticBox';
+
+class EchoOfTheElementals extends Analyzer {
+  procs = 0;
+  damageGained = 0;
+  relevantData=null;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.LAVA_SHOCK.id);
+  }
+
+  on_byPlayer_damage(event) {
+    if (event.ability.guid !== SPELLS.EARTH_SHOCK.id) {
+      return;
+    }
+    const buff = this.selectedCombatant.getBuff(SPELLS.LAVA_SHOCK_BUFF.id);
+    if (buff === undefined) {
+      return;
+    }
+    this.damageGained+= buff.stacks*this.selectedCombatant.traitsBySpellId[SPELLS.LAVA_SHOCK.id];
```

Comment:
Whitespace (and line below)

Also, you need to do a bit more calculation to get the proper damage per stack in case of stacked traits. See here:
https://github.com/WoWAnalyzer/WoWAnalyzer/wiki/%5BBFA%5D-Implementing-Azerite-Traits

Should be done in constructor (if active) and then used to calculate here.